### ISSUE TITLE
ERROR in The target entry-point "listview-directives" has missing dependencies:- ./../

### ISSUE THREAD

**Issue by hariaravind10** [NativeScript/nativescript-angular#2190]
ERROR in The target entry-point "listview-directives" has missing dependencies:- ./../
I am getting the above issue in the latest Nativescript cli and latest angular 9 

My package.json file

{
  "nativescript": {
    "id": "org.nativescript.regalstone",
    "tns-android": {
      "version": "6.5.0"
    },
    "tns-ios": {
      "version": "6.5.0"
    }
  },
  "description": "NativeScript Application",
  "license": "SEE LICENSE IN <your-license-filename>",
  "repository": "<fill-your-repository-here>",
  "scripts": {
    "ngcc": "ngcc --properties es2015 module main --first-only",
    "postinstall": "npm run ngcc"
  },
  "dependencies": {
    "@angular/animations": "~9.1.0",
    "@angular/common": "~9.1.0",
    "@angular/compiler": "~9.1.0",
    "@angular/core": "~9.1.0",
    "@angular/forms": "~9.1.0",
    "@angular/platform-browser": "~9.1.0",
    "@angular/platform-browser-dynamic": "~9.1.0",
    "@angular/router": "~9.1.0",
    "@nativescript/angular": "~9.0.0",
    "@nativescript/schematics": "^9.0.0",
    "@nativescript/theme": "~2.3.0",
    "nativescript-social-share": "^1.6.0",
    "nativescript-ui-listview": "^8.2.0",
    "reflect-metadata": "~0.1.12",
    "rxjs": "^6.5.0",
    "tns-core-modules": "~6.5.0",
    "zone.js": "~0.10.3"
  },
  "devDependencies": {
    "@angular/compiler-cli": "~9.1.0",
    "@ngtools/webpack": "~9.1.0",
    "nativescript-dev-webpack": "~1.5.0",
    "tns-platform-declarations": "~6.5.0",
    "typescript": "~3.8.3"
  },
  "readme": "NativeScript Application"
}
----------------------------------------------------

My ngcc.config.js.   file

module.exports = {
  "packages": {
    "@nativescript/angular": {
      entryPoints: {
        ".": {
          override: {
            main: "./index.js",
            typings: "./index.d.ts",
          },
          ignoreMissingDependencies: true,
        }
      },
      ignorableDeepImportMatchers: [
        /zone.js\//,
        /tns-core-modules\//,
        /@nativescript\/core\//
      ]
    },
    "nativescript-localize": {
      entryPoints: {
        ".": {
          override: {
            main: "./angular.js",
            typings: "./angular.d.ts",
          },
          ignoreMissingDependencies: true,
        },
      },
      ignorableDeepImportMatchers: [
        /@nativescript\/core\//
      ]
    },
    "nativescript-datetimepicker": {
      entryPoints: {
        ".": {
          override: {
            main: "./index.js",
            typings: "./index.d.ts",
          },
          ignoreMissingDependencies: true,
        },
        "angular": {
          override: {
            main: "./index.js",
            typings: "./index.d.ts",
          },
          ignoreMissingDependencies: true,
        }
      },
      ignorableDeepImportMatchers: [
        /tns-core-modules\//,
        /@nativescript\/core\//,
        /@nativescript\/angular\//
      ]
    },
    "nativescript-ui-sidedrawer": {
      entryPoints: {
        "angular": {
          override: {
            main: "./side-drawer-directives.js",
            typings: "./side-drawer-directives.d.ts",
          },
          ignoreMissingDependencies: true,
        }
      },
      ignorableDeepImportMatchers: [
        /tns-core-modules\//,
        /@nativescript\/core\//,
        /@nativescript\/angular\//
      ]
    },
    "nativescript-ui-listview": {
      entryPoints: {
        "angular": {
          override: {
            main: "./listview-directives.js",
            typings: "./listview-directives.d.ts",
          },
          ignoreMissingDependencies: true,
        }
      },
      ignorableDeepImportMatchers: [
        /tns-core-modules\//,
        /@nativescript\/core\//,
        /@nativescript\/angular\//
      ]
    }
  }
}
--------------------------

I got the below error

ERROR in ../node_modules/nativescript-ui-listview/__ivy_ngcc__/angular/listview-directives.js
Module not found: 
Error: Can't resolve './../' in '/Users/Desktop/Aravind/regalstone/node_modules/nativescript-ui-listview/__ivy_ngcc__/angular'
 @ ../node_modules/nativescript-ui-listview/__ivy_ngcc__/angular/listview-directives.js 5:10-26
 @ ./app/app.module.ts
 @ ./main.ts

Please help me

**Comment by funder7** [comment]
```
"tns-android": {
     "version": "6.5.0"
},
"tns-ios": {
     "version": "6.5.0"
}
```

Try to update them to ```6.5.2``` and ```6.5.3``` respectively.

**Comment by juniorschen** [comment]
I went to perform this migration today and the process is hellish, many errors like this because of dependencies, so far things are starting to work for me by osmosis, I have no way to help you but I would like to register my frustration for this update

**Comment by NathanWalker** [comment]
@juniorschen Amen. I find `ngcc` highly frustrating and the update pain Angular puts it's community through would drive anyone insane 🙃  - That being said, we develop a lot of production apps with Angular and NativeScript and are committed to doing what we can do to provide NativeScript compliance with Angular's latest and value the {N} community in the highest regard so will always help in every way we can.

Out of curiosity have you tried just updating to Angular 10?
https://nativescript.org/blog/angular-10-support/

In doing that you can actually switch `nativescript-ui-listview` to "rc" ... Your ngcc file looks fine so leave that as is and can try updating per that blog post. Lemme know.

**Comment by juniorschen** [comment]
@NathanWalker 
I would upgrade to 10 but it's still in the rc version, it would be bad to release it for production, so I decided to go for version 9 for now.

But I'm still looking forward to version 10, I hope support for angular libraries will soon come.

Thanks for your work in nativescrit-angular, however frustrating it may be in an update like this one from ivy, the apps have an incredible performance, and have a very efficient development

**Comment by NathanWalker** [comment]
Totally understand yep. The errors you mention are definitely solvable and if you'd like I'd be happy to help you get unblocked if you'd like to shoot an email to support@nativescript.org - Perhaps a way to add contributor privately temporarily; we've been helping unblock others in similar situation as time permits.

**Comment by NathanWalker** [comment]
@funder7 @lonemanotzi `@nativescript/schematics` **~10.0.0** has now been published alongside `@nativescript/angular` **~10.0.0**

Here's a couple examples if starting fresh (which are already documented) but here for quick reference:

A. Starting a new Angular web project and adding {N} to it afterwards:

```
ng new workspace
cd workspace
ng add @nativescript/schematics --skipAutoGeneratedComponent

// start apps:
npm run ios
npm run android 
```

B. Starting a new Angular NativeScript shared codebase from the start:

```
npm i -g @nativescript/schematics
ng new -c=@nativescript/schematics workspace --shared --style=scss
cd workspace

// start apps:
npm run ios
npm run android 

```

C. If updating existing @nativescript/schematics projects:

1. Delete webpack.config.js (the updated @nativesript/webpack dependency will regenerate a new correct one using latest webpack plugins). If you had customizations, use git change sets to compare the diff.
2. Update package.json as follows (at least in part - can use as a guide):

```
"scripts": {
  "clean": "npx rimraf hooks node_modules package-lock.json platforms",
  "ng": "ng",
  "start": "ng serve",
  "build": "ng build",
  "test": "ng test",
  "lint": "ng lint",
  "e2e": "ng e2e",
  "android": "tns run android --no-hmr",
  "ios": "tns run ios --no-hmr",
  "mobile": "tns run",
  "preview": "tns preview",
  "ngcc": "ngcc --properties es2015 module main --first-only",
  "postinstall": "npm run ngcc"
},
"dependencies": {
  "@angular/animations": "~10.0.0",
  "@angular/common": "~10.0.0",
  "@angular/compiler": "~10.0.0",
  "@angular/core": "~10.0.0",
  "@angular/forms": "~10.0.0",
  "@angular/platform-browser": "~10.0.0",
  "@angular/platform-browser-dynamic": "~10.0.0",
  "@angular/router": "~10.0.0",
  "core-js": "^2.6.9",
  "@nativescript/angular": "~10.0.0",
  "@nativescript/core": "rc",
  "@nativescript/theme": "~2.2.1",
  "reflect-metadata": "~0.1.12",
  "rxjs": "~6.5.5",
  "tslib": "1.10.0",
  "zone.js": "~0.10.2"
},
"devDependencies": {
  "@angular/cli": "~10.0.0",
  "@angular/compiler-cli": "~10.0.0",
  "@angular-devkit/build-angular": "~0.1000.2",
  "@nativescript/tslint-rules": "~0.0.5",
  "@types/jasmine": "~3.5.0",
  "@types/jasminewd2": "~2.0.3",
  "@types/node": "^12.11.1",
  "codelyzer": "^5.1.2",
  "jasmine-core": "~3.5.0",
  "jasmine-spec-reporter": "~4.2.1",
  "karma": "~4.4.1",
  "karma-chrome-launcher": "~3.1.0",
  "karma-coverage-istanbul-reporter": "~2.1.0",
  "karma-jasmine": "~3.0.1",
  "karma-jasmine-html-reporter": "^1.4.2",
  "@nativescript/webpack": "~2.0.0",
  "protractor": "~5.4.3",
  "ts-node": "~8.3.0",
  "tslint": "~6.1.0",
  "typescript": "~3.9.0",
  "node-sass": "^4.7.1"
}
```

3. Clean your projects with `npm run clean` if using the clean scripts from the example scripts above.
